test/typings/either-spec.ts

```diff
@@ -57,7 +57,7 @@ log(unpacked).run();
 log2(getMessage(null).flatMap(getType)).performUnsafeIO();
 log3(getMessage({type: 'x', payload: null}).chain(getType));
 
-const nameError: string = Left('-- Adamovisch').leftMap(n => n.split(' ').shift()).left();
+const nameError: string = Left('-- Adamovisch').leftMap((n:string) => n.split(' ').shift() || "").left();
```

Comment:
yeah, TS type inference is a massive disappointment for me. it's way less good than people think it is. Without the type annotation 'n' is inferred to any. There is another typescript compiler flag that I strongly recommend to enable in general, but for monet, since it's mostly demo code, it's probably not needed. Is it `noImplicitAny`. With that flag, if typescript infers the type of a variable to 'any' but you didn't put any annotation you get a build error. You must give a type, worst-case you type `: any`. I tried to enable that flag for monet and... it wasn't pretty. I think flow's type inference is much better. As it is, typescript really is pretty poor :-(